emote/memory/memory.py

```diff
@@ -463,6 +463,38 @@ def __iter__(self):
             yield {self.data_group: data, self.size_key: data[self.size_key]}
 
 
+class JointMemoryLoader:
+    """A memory loader capable of loading data from multiple `MemoryLoader`s.
+
+    If a datagroup is specified via the `data_group` param, it will place all loaded data in the specified datagroup,
+    otherwise it will directly return the data fetched from each individual loader.
```

Comment:
Ok; hear me out. Subclasses. Default behavior is no data group, a subclass from this that just does the following:

```py
class JointMemoryLoaderWithDataGroup(JointMemoryLoader):
    def __init__(self, ...):
        super().__init__(self)
        self.data_group = ...

    def __iter__(self):
        return { self.data_group: next(super()) }
```
       
Each of these classes becomes more intentional!

Reply:
Do I smell inheritance?? ;) No but sure, I think I like that as well.